packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -706,7 +706,7 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
     }
 
     async executeCommand(commandOptions: CommandLineOptions): Promise<void> {
-        this.sendText(this.shellCommandBuilder.buildCommand(await this.processInfo, commandOptions) + '\n');
+        this.sendText(this.shellCommandBuilder.buildCommand(await this.processInfo, commandOptions) + (OS.backend.isWindows ? '\r\n' : '\n'));
```

Comment:
If I remember correctly, we use this pattern often throughout the application. It might make sense to add `OS.EOL` and `OS.backend.EOL` fields that can be used instead of the ternary operator. That should reduce the probability of these kind of errors.

Reply:
sounds like a good idea